src/bin/ion/commands/generate/generator.rs

```diff
@@ -637,6 +712,27 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
         })
     }
 
+    /// Returns the target type based on given ISL type name and field presence
+    fn target_type_for(
+        field_presence: FieldPresence,
+        name: &String,
+    ) -> Option<FullyQualifiedTypeReference> {
+        let schema_type: IonSchemaType = name.into();
+        L::target_type(&schema_type)
+            .as_ref()
+            .map(|type_name| FullyQualifiedTypeReference {
+                type_name: vec![NamespaceNode::Type(type_name.to_string())],
+                parameters: vec![],
+            })
```

Comment:
This is only going to work if the target type is defined in the same namespace as the location from which it is referenced. If you have e.g. `org.example.util.Fruit` and you are trying to use it in `org.example.SequenceWithImport`, then compilation of the generated java code will fail with an unresolved reference error.

Reply:
Added a warning saying generated code uses a flattened namespace.